stratum/hal/bin/tdi/es2k/es2k_main.cc

```diff
@@ -57,6 +59,13 @@ static void InitCommandLineFlags() {
   // Chassis config file
   SetDefault("chassis_config_file", DEFAULT_CONFIG_DIR "es2k_port_config.pb.txt");
 
+  /* Always create a new file for pipeline_cfg.pb.txt for E2100.
+   * This is required so that previous iteration pipeline config is
+   * applied when infrap4d is restarted
+   */
+  const char *path = FLAGS_forwarding_pipeline_configs_file.c_str();
+  std::ofstream file(path);
```

Comment:
Check for failure?
(also what happens if file exists, does this call overwrite it?)

Reply:
yes this call overwrites any existing file

Reply:
It appears to me that `dpdk_hal.cc` addresses a similar issue (I knew this looked familiar!):
```
  // DPDK cannot configure the pipeline until after the ports have been
  // created, so we ensure that the saved configuration file is empty
  // on startup.
  FILE* pipeline_cfg_file =
    fopen(FLAGS_forwarding_pipeline_configs_file.c_str(), "wb");
  if (pipeline_cfg_file != NULL) {
    LOG(INFO) << "Truncating saved pipeline configuration file.";
    fclose(pipeline_cfg_file);
  }
```
This closer to the code site than the main program is.

Reply:
Ahh okay, do you suggest moving this to es2k_hal.cc for es2k target?

Reply:
Unless there's a compelling reason to do otherwise, I think it is in our best interests to follow the same pattern on both platforms.

Otherwise, we leave a conundrum for the poor maintainer.

Reply:
Moved the logic to es2k_hal.cc